### PR TITLE
feat(ui): data privacy rules selector field with search

### PR DESCRIPTION
**Pull request by priscilawebdev**
**Type:** Feat

**Description:** 

Implemented basic selector autocompletion.. You can read more about it here https://app.asana.com/0/1155078590577881/1166676133180838

Kapture 2020-03-267

### CODE REVIEW

**Comment by untitaker** [comment]
As discussed in hangouts here's my summary.

I found a couple of issues:

- [ ] With an empty textbox, I get suggestions to autocomplete `&&` and `||`, even though that's invalid.
  
Screenshot 2020-03-26 at 12 33 05

- [ ] After typing `!` I need to make a space to get any suggestions. But typing this should give me all suggestions for values: `!$`.

Screenshot 2020-03-26 at 12 33 16

- [ ] When having typed `foo `, I get a suggestion for `!` even though that's invalid.

Screenshot 2020-03-26 at 12 32 57

So I think `!` needs to be treated entirely different from value suggestions and boolean suggestions. I suggest the following naming:

* binary operators `||` and `&&`
* unary operator `!`
* paths/values

This is how we name it in the backend.

Also:

- [ ] I think we need to do something with the tooltips, because right now they just show everything twice when screen space is not an issue. I would suggest to just remove them and ignore the spacing issue on smaller screens.

File: src/sentry/static/sentry/app/components/forms/inputField.tsx [hunk 1]
```diff
@@ -36,6 +41,9 @@ export default class InputField<
         value={this.state.value as string | number} //can't pass in boolean here
         style={this.props.inputStyle}
         onBlur={this.props.onBlur}
+        onKeyPress={this.props.onKeyPress}
+        onFocus={this.props.onFocus}
+        onKeyDown={this.props.onKeyDown}
```
Comment (matejminar) [inline src/sentry/static/sentry/app/components/forms/inputField.tsx]: Does it make sense to destructure these?

File: src/sentry/static/sentry/app/views/settings/components/dataPrivacyRulesPanel/dataPrivacyRulesPanelSelectorField.tsx [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+    const lastFieldValue = fieldValues[fieldValues.length - 1];
+    const penultimateFieldValue = fieldValues[fieldValues.length - 2];
+
+    if (!defined(lastFieldValue)) {
+      return [];
+    }
+
+    if (
+      typeof penultimateFieldValue === 'object' &&
+      penultimateFieldValue.type === 'boolean'
```
Comment (matejminar) [inline src/sentry/static/sentry/app/views/settings/components/dataPrivacyRulesPanel/dataPrivacyRulesPanelSelectorField.tsx]: Can we do something like `penultimateFieldValue?.type === 'boolean'` without checking if it's of type object first?

File: src/sentry/static/sentry/app/views/settings/components/dataPrivacyRulesPanel/dataPrivacyRulesPanelSelectorField.tsx [hunk 3]
```diff
@@ -0,0 +1,283 @@
...
+    ) {
+      return valueSelectors;
+    }
+
+    if (fieldValues.length > 1 && typeof lastFieldValue === 'string' && !lastFieldValue) {
+      return booleanSelectors;
+    }
+
+    const value =
+      typeof lastFieldValue === 'object' ? lastFieldValue.value : lastFieldValue;
```
Comment (matejminar) [inline src/sentry/static/sentry/app/views/settings/components/dataPrivacyRulesPanel/dataPrivacyRulesPanelSelectorField.tsx]: Should we do `lastFieldValue.value ?? lastFieldValue ` instead?

File: src/sentry/static/sentry/app/views/settings/components/dataPrivacyRulesPanel/dataPrivacyRulesPanelSelectorField.tsx [hunk 4]
```diff
@@ -0,0 +1,283 @@
...
+
+    const filteredSuggestions = selectors.filter(
+      s => s.value.indexOf(value.toLowerCase()) > -1
```
Comment (matejminar) [inline src/sentry/static/sentry/app/views/settings/components/dataPrivacyRulesPanel/dataPrivacyRulesPanelSelectorField.tsx]: Shoud we use `s.value.includes`?

File: src/sentry/static/sentry/app/views/settings/components/dataPrivacyRulesPanel/dataPrivacyRulesPanelSelectorFieldTypes.tsx [hunk 5]
```diff
@@ -0,0 +1,105 @@
+const valueSelectors: Suggestions = [
+  {
+    type: 'value',
+    value: '$string',
+    description: 'Any string value',
```
Comment (matejminar) [inline src/sentry/static/sentry/app/views/settings/components/dataPrivacyRulesPanel/dataPrivacyRulesPanelSelectorFieldTypes.tsx]: Should all descriptions in this file be translated?

**Review by matejminar — COMMENTED** [review]
Good job! 🚀
I just left a few comments to discuss.

File: src/sentry/static/sentry/app/views/settings/components/dataPrivacyRulesPanel/dataPrivacyRulesPanelSelectorField.tsx [hunk 6]
```diff
@@ -0,0 +1,283 @@
...
+              >
+                <Tooltip
+                  title={`${suggestion.value} ${suggestion?.description &&
+                    `(${suggestion.description})`}`}
+                  position="top"
+                >
+                  <SuggestionItem>
+                    <TextOverflow>{suggestion.value}</TextOverflow>
+                    {suggestion?.description && (
+                      <SuggestionDescription>
```
Comment (untitaker) [inline src/sentry/static/sentry/app/views/settings/components/dataPrivacyRulesPanel/dataPrivacyRulesPanelSelectorField.tsx]: @doralchan was suggesting to make the description gray.

File: src/sentry/static/sentry/app/views/settings/components/dataPrivacyRulesPanel/dataPrivacyRulesPanelSelectorField.tsx (hunk 3 above)
Comment (priscilawebdev) [reply src/sentry/static/sentry/app/views/settings/components/dataPrivacyRulesPanel/dataPrivacyRulesPanelSelectorField.tsx]: not really 
image

File: src/sentry/static/sentry/app/views/settings/components/dataPrivacyRulesPanel/dataPrivacyRulesPanelSelectorField.tsx (hunk 6 above)
Comment (priscilawebdev) [reply src/sentry/static/sentry/app/views/settings/components/dataPrivacyRulesPanel/dataPrivacyRulesPanelSelectorField.tsx]: Good idea 💡

File: src/sentry/static/sentry/app/views/settings/components/dataPrivacyRulesPanel/dataPrivacyRulesPanelSelectorField.tsx (hunk 2 above)
Comment (priscilawebdev) [reply src/sentry/static/sentry/app/views/settings/components/dataPrivacyRulesPanel/dataPrivacyRulesPanelSelectorField.tsx]: not really, but I will check if I can improve this part of the code

**Review by untitaker — APPROVED** [review]
tried it out, lgtm!

**Pull request merged**